src/Tearing/TearingEngine.inl

```diff
@@ -894,96 +903,73 @@ void TearingEngine<DataTypes>::handleEvent(sofa::core::objectmodel::Event* event
         {
             fractureSegmentEndpoints.push_back(Pb);
             fractureSegmentEndpoints.push_back(Pc);
-
         }
     }
-  
-    if (/* simulation::AnimateBeginEvent* ev = */simulation::AnimateEndEvent::checkEventType(event))
+
+    // Perform fracture every d_stepModulo
+    int step = d_stepModulo.getValue();
+    if (step == 0) // interactive version
     {
-        int step = d_stepModulo.getValue();
-        if (step == 0) // interactive version
-        {
-            if (m_stepCounter > 200 && (m_tearingAlgo->getFractureNumber() < d_nbFractureMax.getValue()))
-                algoFracturePath();
-        }
-        else if (((m_stepCounter % step) == 0) && (m_tearingAlgo->getFractureNumber() < d_nbFractureMax.getValue()))
-        {
-            if (m_stepCounter > d_stepModulo.getValue())
-                algoFracturePath();
-        }
+        if (m_stepCounter > 200 && (m_tearingAlgo->getFractureNumber() < d_nbFractureMax.getValue()))
+            algoFracturePath();
     }
-
-    if (sofa::core::objectmodel::KeypressedEvent* ev = dynamic_cast<sofa::core::objectmodel::KeypressedEvent*>(event))
+    else if (((m_stepCounter % step) == 0) && (m_tearingAlgo->getFractureNumber() < d_nbFractureMax.getValue()))
     {
-        if (ev->getKey() == 'C')
-        {
+        if (m_stepCounter > d_stepModulo.getValue())
```

Comment:
- This if statement is always true, since (m_stepCounter % step) == 0) will always be true when m_stepCounter is divisible by step. So, this if statement is redundant.